projects/goharbor/harbor/Makefile

```diff
@@ -0,0 +1,100 @@
+HARBOR_REDIS_IMAGE_COMPONENT=harbor/harbor-redis
+HARBOR_EXPORTER_IMAGE_COMPONENT=harbor/harbor-exporter
+
+IMAGE_NAMES=harbor-db harbor-portal harbor-core harbor-log harbor-nginx harbor-jobservice harbor-registry harbor-registryctl harbor-redis harbor-exporter
+
+BINARY_TARGET_FILES=harbor-core harbor-jobservice harbor-registry harbor-registryctl harbor-exporter
+SOURCE_PATTERNS=./core ./jobservice ./registry ./registryctl ./cmd/exporter
+EXTRA_GOBUILD_FLAGS=-mod=vendor
+
+$(call IMAGE_TARGETS_FOR_NAME, harbor-db): harbor-db-useradd/images/export
+$(call IMAGE_TARGETS_FOR_NAME, harbor-portal): harbor-portal-useradd/images/export
+$(call IMAGE_TARGETS_FOR_NAME, harbor-core): harbor-core-useradd/images/export
+$(call IMAGE_TARGETS_FOR_NAME, harbor-log): harbor-log-useradd/images/export
+$(call IMAGE_TARGETS_FOR_NAME, harbor-nginx): harbor-nginx-useradd/images/export
+$(call IMAGE_TARGETS_FOR_NAME, harbor-jobservice): harbor-jobservice-useradd/images/export
+$(call IMAGE_TARGETS_FOR_NAME, harbor-registry): harbor-registry-useradd/images/export
+$(call IMAGE_TARGETS_FOR_NAME, harbor-registryctl): harbor-registryctl-useradd/images/export
+$(call IMAGE_TARGETS_FOR_NAME, harbor-redis): harbor-redis-useradd/images/export
+$(call IMAGE_TARGETS_FOR_NAME, harbor-exporter): harbor-exporter-useradd/images/export
+
+harbor-db-useradd/images/export: IMAGE_USERADD_USER_NAME=postgres IMAGE_USERADD_USER_ID=999
+harbor-portal-useradd/images/export: IMAGE_USERADD_USER_NAME=nginx IMAGE_USERADD_USER_ID=10000
+harbor-core-useradd/images/export: IMAGE_USERADD_USER_NAME=harbor IMAGE_USERADD_USER_ID=10000
+harbor-log-useradd/images/export: IMAGE_USERADD_USER_NAME=syslog IMAGE_USERADD_USER_ID=10000
+harbor-nginx-useradd/images/export: IMAGE_USERADD_USER_NAME=nginx IMAGE_USERADD_USER_ID=10000
+harbor-jobservice-useradd/images/export: IMAGE_USERADD_USER_NAME=harbor IMAGE_USERADD_USER_ID=10000
+harbor-registry-useradd/images/export: IMAGE_USERADD_USER_NAME=harbor IMAGE_USERADD_USER_ID=10000
+harbor-registryctl-useradd/images/export: IMAGE_USERADD_USER_NAME=harbor IMAGE_USERADD_USER_ID=10000
+harbor-redis-useradd/images/export: IMAGE_USERADD_USER_NAME=redis
+harbor-exporter-useradd/images/export: IMAGE_USERADD_USER_NAME=harbor IMAGE_USERADD_USER_ID=10000
+
+PRECOMPILE_PREPARE=harbor_prepare
+
+DOCKERFILEPATH=$(PROJECT_ROOT)/$(REPO)/make/photon
+DOCKERFILEPATH_REG=$(DOCKERFILEPATH)/registry
+DOCKERFILEPATH_CHART_SERVER=$(DOCKERFILEPATH)/chartserver
+MAKE_PREPARE_PATH=$(DOCKERFILEPATH)/prepare
+
+harbor_base_namespace=goharbor
+IMAGE_BUILD_ARGS=harbor_base_namespace
+
+harbor_prepare: version_prepare swagger_prepare registry_prepare
+
+export VERSIONS_FOR_PREPARE
+version_prepare:
+	@echo "$$VERSIONS_FOR_PREPARE" > $(MAKE_PREPARE_PATH)/versions
+
+swagger_prepare:
+	make -C $(REPO) gen_apis
+
+registry_prepare:
+	cd $(PROJECT_ROOT)/$(REPO)/$(REPO_SUBPATH) \
+	&& git clone -b $(REGISTRY_SRC_TAG) https://github.com/distribution/distribution.git registry && cd registry \
```

Comment:
registry compiles when I put its source code along with harbor main source code. But I can pull that out into a new project and make it as a dependency.

Reply:
registry has been factored out to a new project distribution/distribution